nix/flake-module.nix

```diff
@@ -125,14 +125,15 @@ in
                   overrides, on top of `basePackages`.
                 '';
               };
-              localPackages = mkOption {
-                type = types.attrsOf types.package;
+              packages = mkOption {
+                # TODO: use a more specific type
+                type = types.attrsOf types.raw;
```

Comment:
`types.submodule`?

Reply:
`outputs.package` is an attrset where attribute keys are `<packageName>`. The closest possible type I see is `types.attrsOf types.attrs`. What do you think?

Reply:
@sbh69840 It should be like this:

https://github.com/srid/haskell-flake/blob/8fb3bbe77c74f2b7249d769d2eb462df917bbdc1/nix/flake-module.nix#L290

which is then defined as:

https://github.com/srid/haskell-flake/blob/8fb3bbe77c74f2b7249d769d2eb462df917bbdc1/nix/flake-module.nix#L155-L161

So you are looking for something like:

```nix
let 
  packageInfoSubmodule = types.submoduleWith {
    modules = [
      ({ config, name, ...}: {
        # 'name' refers to package name
        options = ...
      })
    ];
  };
in
  options.outputs.packages = mkOption { type = packageInfoSubmodule; ...
```